### DISCUSSION TITLE
Service 'fixed charge/discharge periods' has no effect?

### DISCUSSION THREAD

**Discussion by timconinx** [wlcrs/huawei_solar#649] [General]
### Describe the issue

Setting my battery in 'Fixed Charge Discharge' mode, I tried to use the service 'Set Fixed Charge/Discharge Periods' to enable the battery to charge and discharge at fixed times during the day.

The service itself works insofar I got errors saying the times weren't formatted correctly (using `3:30` instead of `03:30`) but after fixing those there were no errors and the attributed showed correctly in HA
![afbeelding](https://github.com/wlcrs/huawei_solar/assets/1834193/ebcfe2a6-ba33-48c3-9d73-77c36469f5cb)

However, no charging was done and nothing appears in the logs. I didn't change the battery's 'Fixed Charge Discharge' mode, power didn't exceed the global 'Grid Charge Maximum Power' variable... there was just no action from the battery

I did notice that nor 'Fixed Charge/Discharge Periods' setting, nor the service to change it are mentioned in the wiki documentation... did I try something too experimental?

### Bescribe your Huawei Solar Setup

Inverter Type: SUN2000-4KTL-M1
Inverter Firmware version: V100R001C00SPC153
SDongle present: yes
Power meter present: three phase
Battery: LUNA2000 10kWh
Battery Firmware version: V100R002C00SPC123 

### How do you connect to the inverter?

Via the SDongle, wired connection

### Upload your Diagnostics File

Drag & Drop your Diagnostics File here.
[config_entry-huawei_solar-399774178e4698d097ea7183f3a5fe75.json](https://github.com/wlcrs/huawei_solar/files/14555191/config_entry-huawei_solar-399774178e4698d097ea7183f3a5fe75.json)

### Upload your relevant debug logs

```text
There are no relevant logs
```

### Please confirm the following:

- [X] I'm running the latest release of Home Assistant.
- [X] I'm running the latest release of this integration.
- [X] I did not find an existing issue describing this problem.
- [X] I did upload the diagnostics-file that I could retrieve from  the 'Devices & Services Page'
- [X] I did upload the relevant debug logs (via 'Enable Debug Logging'-feature or by manually configuring HA logging)

**Comment by wlcrs** [comment]
The integration successfully writes the TOU configuration to the inverter. How the inverter and battery act on that configuration is out of scope for this integration, and I do not provide any support for this whatsoever.

I'm converting this into a discussion, hoping that other users can help you with this.

**Reply by timconinx** [reply]
Thanks @wlcrs 

Just as a clarification: TOU periods (set by `set fixed TOU periods` ) absolutely work. Fixed Charge/Discharge periods do not. I agree that this is a matter of how the inverter/battery works with the difference between the two.

Here's hoping other people are experimenting as well with these new(?) features

**Reply by mk2lehe** [reply]
I just started playing with this, since it may be useful for my EMHASS setup. 
I can confirm that something does not work as intended, reproduced by Developer tools -> Actions -> Huawei Solar: Set Fixed Charge Periods, choose the battery, then just copy and paste the line in the example (12:00-15:59/-1000W) gives Unknown error. However, without the negative sign it gets accepted (12:00-15:59/1000W).
